lib/home/home_page.dart

```diff
@@ -23,38 +25,100 @@ class HomePage extends StatelessWidget {
   }
 }
 
-class HomeView extends StatelessWidget {
+class HomeView extends StatefulWidget {
   const HomeView({super.key});
 
+  @override
+  State<HomeView> createState() => _HomeViewState();
+}
+
+class _HomeViewState extends State<HomeView> {
+  late ScrollController _scrollController;
+  double expandedHeight = 132;
+  bool lastStatus = true;
+
+  bool get _isShrunk {
+    return _scrollController.hasClients &&
+        _scrollController.offset > (expandedHeight - kToolbarHeight);
+  }
+
+  void _scrollListener() {
+    if (_isShrunk != lastStatus) {
+      setState(() {
+        lastStatus = _isShrunk;
+      });
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController = ScrollController()..addListener(_scrollListener);
+  }
+
   @override
   Widget build(BuildContext context) {
     return CustomScrollView(
+      controller: _scrollController,
       slivers: [
         SliverAppBar(
```

Comment:
I would be nice if this SliverAppBar could have its dedicated widget, so that the HomePage could stay Stateless